lib/cache/index.ts

```diff
@@ -0,0 +1,68 @@
+import { CacheOptions, DEFAULT_OPTIONS } from './CacheOptions';
+import { cacheFactory } from './factories/cacheFactory';
+
+export { CacheOptions };
+
+interface CacheOptionsAndTimeout extends CacheOptions {
```

Comment:
I'd suggest better to rename it to `TimeoutCacheOptions`. At the same time `CacheOptions` is a type, why making this one an interface and not type (for consistency)?